### PR TITLE
Feature/table-refactor

### PR DESCRIPTION
**Pull request by DanSallau**

### CODE REVIEW

File: src/Table/sections/helperFunctions.ts [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    if (useSelection) {
+        sum = sum + 1;
+    }
+
+    if (useGroupBy) {
+        sum = sum + 1;
+    }
+    if (useShowActionColumn) {
+        sum = sum + 1;
+    }
```
Comment (yousifalraheem) [inline src/Table/sections/helperFunctions.ts:22]:
```suggestion
    if (useSelection) {
        sum++;
    }

    if (useGroupBy) {
        sum++;
    }

    if (useShowActionColumn) {
        sum++;
    }
```

File: src/Table/sections/helperFunctions.ts [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+/**
+ * sort array of tabke rows
+ * @param items table rows array
+ * @param columnName the target column name
+ * @param sortDirection the sort direction
+ * @return Array of tableRow
+ */
+export function sortArray(items: Array<TableRow> = [], columnName: string, sortDirection: sortDirectionTypes): Array<TableRow> {
+    const languages: Readonly<Array<string>> = window.navigator?.languages || ["sw", "en"];
```
Comment (yousifalraheem) [inline src/Table/sections/helperFunctions.ts:35]: I don't think you should force `sw` and `en` on the component. You should find a way to get the language from the `navigator` or any other way.

Comment (DanSallau) [reply src/Table/sections/helperFunctions.ts:35]: we are getting from navigator "sw" and "en" are just a fallback incase users browser doesn't support one.

**Pull request merged**